File: benchmark/fib.lua
function fib(n)
  if n < 2 then return n end
  return fib(n - 2) + fib(n - 1)
end

local start = os.clock()
for i = 1, 5 do
  io.write(fib(28) .. "\n")
end
io.write(string.format("elapsed: %.8f\n", os.clock() - start))

File: benchmark/fib.js
function fib(n) {
  if (n < 2) return n;
  return fib(n - 1) + fib(n - 2);
}

var start = process.hrtime();
var i = 0;
for (var i = 0; i < 5; i++) {
  console.log(fib(30));
}
var elapsed = process.hrtime(start);
elapsed = elapsed[0] + elapsed[1] / 1000000000;
console.log("elapsed: " + elapsed);
